main.py

```diff
@@ -8,17 +8,37 @@ def train(train_data, model):
     """
     the train function, train the model for an entire epoch
 
-    :param train_data: the preprocessed training data, of shape [num_data_points, state_size]
+    :param train_data: the preprocessed training data, of shape [num_stocks, state_size, num_days]
     :param model: the RL agent
 
     :return to be decided
     """
     batch_size = model.batch_size
+    num_days = train_data.shape[2]
     # TODO: 0) batch the train_data, and for each batch:
     # TODO: 1) generate an episode, put it in the memory buffer (use generate_episode() from stock_env.py)
     # TODO: 2) sample a batch of (state, action, reward) from the memory buffer
     # TODO: 3) compute the discounted rewards
     # TODO: 4) compute the loss, run back prop on the model
+    for batch in range(0, int(num_days / batch_size)):
+        print("Training batch #{}".format(batch))
+        start = batch * batch_size
+        end = start + batch_size
+        batch_input = train_data[:, :, start:end]
+        with tf.GradientTape() as tape:
+            states, actions, rewards = generate_episode(batch_input, model)
+            discounted_rewards = discount(rewards)
+            model.buffer.append((states, actions, discounted_rewards))
```

Comment:
the buffer is actually arranged a tiny bit different than this, see `agent.py`'s `remember()` .
This should call `model.remember(states, actions, discounted_rewards))`